### PR TITLE
refactor(providers): Migrate from onesignal-node SDK to the official @onesignal/node-onesignal

### LINKED ISSUE #4635: [NV-3053] Migrate from onesignal-node SDK to the official @onesignal/node-onesignal
## What?
In the @novu/one-signal package we would like to migrate from using the onesignal-node SDK to the official @onesignal/node-onesignal SDK. Check the documentation about the it here: https://github.com/OneSignal/onesignal-node-api#readme

## Why? (Context)
Due to vulnerabilities and un-maintained SDK, we should move from using unofficial onesignal-node SDK.

## Definition of Done
- the onesignal-node SDK package has been replaced with @onesignal/node-onesignal
- the current provider functionality works with the calls to the new @onesignal/node-onesignal SDK:
  - attachments
  - email delivery with all fields

### PR DESCRIPTION
**Pull request by Dhoni77**
### What change does this PR introduce?

- Remove onesignal-node sdk
- Use `@onesignal/node-onesignal` sdk
- Updated unit tests

### Why was this change needed?

Closes #4635

Due to vulnerabilities and un-maintained SDK, we should move from using unofficial onesignal-node SDK.

### CODE REVIEW

File: packages/application-generic/src/factories/push/handlers/one-signal.handler.ts [hunk 1]
```diff
@@ -8,13 +8,18 @@ export class OneSignalHandler extends BasePushHandler {
...
+      !credentials.appKey
+    ) {
       throw Error('Config is not valid for OneSignal');
     }
 
     this.provider = new OneSignalPushProvider({
       appId: credentials.applicationId,
-      apiKey: credentials.apiKey,
+      appKey: credentials.appKey,
+      userKey: credentials.userKey,
```
Comment (scopsy) [inline packages/application-generic/src/factories/push/handlers/one-signal.handler.ts:22]: Unfortunately this change is causing a breakin change for all of one signal users, is there some mapping between the old apiKey to the new appKey? is that the same value?

**Review by scopsy — CHANGES_REQUESTED** [review]
We need to make sure that we are not providing a breaking change for old configuration users who used the apiKey and applicationId approach.

**Comment by scopsy** [comment]
@Dhoni77 another approach we can take is using the rest API directly which works with API key and AppId like previously: https://documentation.onesignal.com/reference/create-notification we can use axios for the API call, wdyt?

File: providers/one-signal/package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
   },
   "dependencies": {
     "@novu/stateless": "^0.21.0",
-    "onesignal-node": "^3.4.0"
+    "axios": "1.3.4"
```
[the remark this answers is not in the document]
Comment (Dhoni77) [reply providers/one-signal/package.json:34]: Ok👍

File: providers/one-signal/src/lib/one-signal.provider.spec.ts [hunk 3]
```diff
@@ -1,56 +1,50 @@
 import { OneSignalPushProvider } from './one-signal.provider';
+import axios from 'axios';
+import { IPushOptions } from 'packages/stateless/build/main';
```
Comment (scopsy) [inline providers/one-signal/src/lib/one-signal.provider.spec.ts:3]:
```suggestion
import { IPushOptions } from '@novu/stateless';
```

**Review by scopsy — APPROVED** [review]

**Pull request merged**